src/main/kotlin/com/cosmotech/api/utils/SecurityUtils.kt

```diff
@@ -43,8 +58,15 @@ fun getCurrentAuthenticatedRoles(configuration: CsmPlatformProperties): List<Str
     throw IllegalStateException("User Authentication not found in Security Context")
   }
 
-  val authentication = getCurrentAuthentication() as BearerTokenAuthentication
-  return authentication.token.tokenValue.let {
+  val authentication = getCurrentAuthentication()
+
+  if (authentication is JwtAuthenticationToken) {
+    return authentication.token.tokenValue.let {
+      JWTParser.parse(it).jwtClaimsSet.getStringListClaim(configuration.authorization.rolesJwtClaim)
```

Comment:
The same, like before, line 2 times: `JWTParser.parse(it).jwtClaimsSet.getStringListClaim(configuration.authorization.rolesJwtClaim)`